11:34:13.898 karate.env system property was:  
11:34:14.974 request:
1 > POST http://127.0.0.1:5000/users
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Content-Length: 63
1 > Content-Type: application/json; charset=UTF-8
1 > Host: 127.0.0.1:5000
1 > User-Agent: Apache-HttpClient/4.5.11 (Java/1.8.0_241)
{"name":"harley","email":"harley@gmail.com","password":"henry"}

11:34:15.685 response time in milliseconds: 709.34
1 < 200
1 < Content-Length: 25
1 < Content-Type: text/html; charset=utf-8
1 < Date: Tue, 31 Mar 2020 06:04:15 GMT
1 < Server: Werkzeug/0.16.1 Python/3.6.6
User Created Successfully

equest failed: 
org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:5000 [/127.0.0.1] failed: Connection refused: connect

    "email": "john@gmail.com", 
    "id": 6, 
    "name": "John Smith", 
    "password": "balaji"
  }, 
  {
    "email": "jack@gmail.com", 
    "id": 8, 
    "name": "jack", 
    "password": "jack"
  }, 
  {
    "email": "john@gmail.com", 
    "id": 9, 
    "name": "john", 
    "password": "john"
  }, 
  {
    "email": "mary@gmail.com", 
    "id": 13, 
    "name": "mary", 
    "password": "mary"
  }, 
  {
    "email": "jackson@gmail.com", 
    "id": 14, 
    "name": "jackson", 
    "password": "jackson"
  }, 
  {
    "email": "henry@gmail.com", 
    "id": 15, 
    "name": "henry", 
    "password": "henry"
  }, 
  {
    "email": "magrit@gmail.com", 
    "id": 17, 
    "name": "magrit", 
    "password": "henry"
  }, 
  {
    "email": "harley@gmail.com", 
    "id": 18, 
    "name": "harley", 
    "password": "henry"
  }, 
  {
    "email": "harley@gmail.com", 
    "id": 19, 
    "name": "harley", 
    "password": "henry"
  }, 
  {
    "email": "harley@gmail.com", 
    "id": 20, 
    "name": "harley", 
    "password": "henry"
  }
]


iseconds for URL: http://127.0.0.1:5000/users
22:31:34.368 http request failed: 
org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:5000 [/127.0.0.1] failed: Connection refused: connect
22:31:34.390 karate.env system property was:  
22:31:34.408 request:
1 > GET http://127.0.0.1:5000/user/2
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Host: 127.0.0.1:5000
1 > User-Agent: Apache-HttpClient/4.5.11 (Java/1.8.0_241)

22:31:35.422 org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:5000 [/127.0.0.1] failed: Connection refused: connect, http call failed after 1013 milliseconds for URL: http://127.0.0.1:5000/user/2
22:31:35.422 http request failed: 
org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:5000 [/127.0.0.1] failed: Connection refused: connect
22:31:35.458 karate.env system property was:  
22:31:35.464 request:
1 > PUT http://127.0.0.1:5000/user/13
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Content-Length: 58
1 > Content-Type: application/json; charset=UTF-8
1 > Host: 127.0.0.1:5000
1 > User-Agent: Apache-HttpClient/4.5.11 (Java/1.8.0_241)
{"name":"mary","email":"mary@gmail.com","password":"mary"}

22:31:36.468 org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:5000 [/127.0.0.1] failed: Connection refused: connect, http call failed after 1003 milliseconds for URL: http://127.0.0.1:5000/user/13
22:31:36.469 http request failed: 
org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:5000 [/127.0.0.1] failed: Connection refused: connect
pt exc.NO_STATE as err:</pre>
<pre class="line after"><span class="ws">            </span>util.raise_(</pre>
<pre class="line after"><span class="ws">                </span>exc.UnmappedInstanceError(instance), replace_context=err,</pre>
<pre class="line after"><span class="ws">            </span>)</pre>
<pre class="line after"><span class="ws"></span> </pre></div>
</div>

<li><div class="exc-divider">The above exception was the direct cause of the following exception:</div>
<li><div class="frame" id="frame-2371606791000">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">2463</em>,
      in <code class="function">__call__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>def __call__(self, environ, start_response):</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;The WSGI server calls the Flask application object as the</pre>
<pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app` which can be</pre>
<pre class="line before"><span class="ws">        </span>wrapped to applying middleware.&quot;&quot;&quot;</pre>
<pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def __repr__(self):</pre>
<pre class="line after"><span class="ws">        </span>return &quot;&lt;%s %r&gt;&quot; % (self.__class__.__name__, self.name)</pre></div>
</div>

<li><div class="frame" id="frame-2371606791784">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">2449</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line before"><span class="ws">                </span>error = e</pre>
<pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre></div>
</div>

<li><div class="frame" id="frame-2371606792064">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1866</em>,
      in <code class="function">handle_exception</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span># if we want to repropagate the exception, we can attempt to</pre>
<pre class="line before"><span class="ws">            </span># raise it with the whole traceback in case we can do that</pre>
<pre class="line before"><span class="ws">            </span># (the function was actually called from the except part)</pre>
<pre class="line before"><span class="ws">            </span># otherwise, we just raise the error again</pre>
<pre class="line before"><span class="ws">            </span>if exc_value is e:</pre>
<pre class="line current"><span class="ws">                </span>reraise(exc_type, exc_value, tb)</pre>
<pre class="line after"><span class="ws">            </span>else:</pre>
<pre class="line after"><span class="ws">                </span>raise e</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>self.log_exception((exc_type, exc_value, tb))</pre>
<pre class="line after"><span class="ws">        </span>server_error = InternalServerError()</pre></div>
</div>

<li><div class="frame" id="frame-2371606793688">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py"</cite>,
      line <em class="line">39</em>,
      in <code class="function">reraise</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>import collections.abc as collections_abc</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>def reraise(tp, value, tb=None):</pre>
<pre class="line before"><span class="ws">        </span>if value.__traceback__ is not tb:</pre>
<pre class="line before"><span class="ws">            </span>raise value.with_traceback(tb)</pre>
<pre class="line current"><span class="ws">        </span>raise value</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>implements_to_string = _identity</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>else:</pre>
<pre class="line after"><span class="ws">    </span>iterkeys = lambda d: d.iterkeys()</pre></div>
</div>

<li><div class="frame" id="frame-2371606793464">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">2446</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
<pre class="line before"><span class="ws">        </span>error = None</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">                </span>error = e</pre>
<pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre></div>
</div>

<li><div class="frame" id="frame-2371606792288">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1951</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(self, rv, from_error_handler=False):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Given the return value from a view function this finalizes</pre>
<pre class="line after"><span class="ws">        </span>the request by converting it into a response and invoking the</pre></div>
</div>

<li><div class="frame" id="frame-2371606790832">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1820</em>,
      in <code class="function">handle_user_exception</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>return self.handle_http_exception(e)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>handler = self._find_error_handler(e)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>if handler is None:</pre>
<pre class="line current"><span class="ws">            </span>reraise(exc_type, exc_value, tb)</pre>
<pre class="line after"><span class="ws">        </span>return handler(e)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def handle_exception(self, e):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Handle an exception that did not have an error handler</pre>
<pre class="line after"><span class="ws">        </span>associated with it, or that was raised from an error handler.</pre></div>
</div>

<li><div class="frame" id="frame-2371606792960">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py"</cite>,
      line <em class="line">39</em>,
      in <code class="function">reraise</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>import collections.abc as collections_abc</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>def reraise(tp, value, tb=None):</pre>
<pre class="line before"><span class="ws">        </span>if value.__traceback__ is not tb:</pre>
<pre class="line before"><span class="ws">            </span>raise value.with_traceback(tb)</pre>
<pre class="line current"><span class="ws">        </span>raise value</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>implements_to_string = _identity</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>else:</pre>
<pre class="line after"><span class="ws">    </span>iterkeys = lambda d: d.iterkeys()</pre></div>
</div>

<li><div class="frame" id="frame-2371606793352">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1949</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>self.try_trigger_before_first_request_functions()</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line current"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(self, rv, from_error_handler=False):</pre></div>
</div>

<li><div class="frame" id="frame-2371606790384">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1935</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>getattr(rule, &quot;provide_automatic_options&quot;, False)</pre>
<pre class="line before"><span class="ws">            </span>and req.method == &quot;OPTIONS&quot;</pre>
<pre class="line before"><span class="ws">        </span>):</pre>
<pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
<pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
<pre class="line current"><span class="ws">        </span>return self.view_functions[rule.endpoint](**req.view_args)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def full_dispatch_request(self):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Dispatches the request and on top of that performs request</pre>
<pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
<pre class="line after"><span class="ws">        </span>error handling.</pre></div>
</div>

<li><div class="frame" id="frame-2371606791952">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py"</cite>,
      line <em class="line">89</em>,
      in <code class="function">view</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>constructor of the class.</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>def view(*args, **kwargs):</pre>
<pre class="line before"><span class="ws">            </span>self = view.view_class(*class_args, **class_kwargs)</pre>
<pre class="line current"><span class="ws">            </span>return self.dispatch_request(*args, **kwargs)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>if cls.decorators:</pre>
<pre class="line after"><span class="ws">            </span>view.__name__ = name</pre>
<pre class="line after"><span class="ws">            </span>view.__module__ = cls.__module__</pre>
<pre class="line after"><span class="ws">            </span>for decorator in cls.decorators:</pre></div>
</div>

<li><div class="frame" id="frame-2371606792624">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py"</cite>,
      line <em class="line">163</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span># retry with GET.</pre>
<pre class="line before"><span class="ws">        </span>if meth is None and request.method == &quot;HEAD&quot;:</pre>
<pre class="line before"><span class="ws">            </span>meth = getattr(self, &quot;get&quot;, None)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>assert meth is not None, &quot;Unimplemented method %r&quot; % request.method</pre>
<pre class="line current"><span class="ws">        </span>return meth(*args, **kwargs)</pre></div>
</div>

<li><div class="frame" id="frame-2371606792008">
  <h4>File <cite class="filename">"C:\Users\admin\PycharmProjects\TrainingTask\app.py"</cite>,
      line <em class="line">112</em>,
      in <code class="function">delete</code></h4>
  <div class="source "><pre class="line before"><span class="ws">                  </span>type: array</pre>
<pre class="line before"><span class="ws">                  </span>items:</pre>
<pre class="line before"><span class="ws">                    </span>$ref: '#/definitions/User'</pre>
<pre class="line before"><span class="ws">                </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws">        </span>data = UserModel.query.get(user_id)</pre>
<pre class="line current"><span class="ws">        </span>db.session.delete(data)</pre>
<pre class="line after"><span class="ws">        </span>db.session.commit()</pre>
<pre class="line after"><span class="ws">        </span>return jsonify(UserModelSchema().dump(data))</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def put(self, user_id):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;</pre></div>
</div>

<li><div class="frame" id="frame-2371606791392">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\scoping.py"</cite>,
      line <em class="line">162</em>,
      in <code class="function">do</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span>&quot;&quot;&quot;Old name for backwards compatibility.&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws"></span>def instrument(name):</pre>
<pre class="line before"><span class="ws">    </span>def do(self, *args, **kwargs):</pre>
<pre class="line current"><span class="ws">        </span>return getattr(self.registry(), name)(*args, **kwargs)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>return do</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>for meth in Session.public_methods:</pre></div>
</div>

<li><div class="frame" id="frame-2371606791448">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py"</cite>,
      line <em class="line">2015</em>,
      in <code class="function">delete</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>state = attributes.instance_state(instance)</pre>
<pre class="line before"><span class="ws">        </span>except exc.NO_STATE as err:</pre>
<pre class="line before"><span class="ws">            </span>util.raise_(</pre>
<pre class="line current"><span class="ws">                </span>exc.UnmappedInstanceError(instance), replace_context=err,</pre>
<pre class="line after"><span class="ws">            </span>)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>self._delete_impl(state, instance, head=True)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def _delete_impl(self, state, obj, head):</pre></div>
</div>

<li><div class="frame" id="frame-2371606790496">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\util\compat.py"</cite>,
      line <em class="line">178</em>,
      in <code class="function">raise_</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span># intermediary exception does not change, but we can't figure</pre>
<pre class="line before"><span class="ws">            </span># that out.</pre>
<pre class="line before"><span class="ws">            </span>exception.__cause__ = replace_context</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line current"><span class="ws">            </span>raise exception</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre>
<pre class="line after"><span class="ws">            </span># credit to</pre>
<pre class="line after"><span class="ws">            </span># https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/</pre>
<pre class="line after"><span class="ws">            </span># as the __traceback__ object creates a cycle</pre>
<pre class="line after"><span class="ws">            </span>del exception, replace_context, from_, with_traceback</pre></div>
</div>
</ul>
  <blockquote>sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped</blockquote>
</div>

<div class="plain">
  <form action="/?__debugger__=yes&amp;cmd=paste" method="post">
    <p>
      <input type="hidden" name="language" value="pytb">
      This is the Copy/Paste friendly version of the traceback.  <span
      class="pastemessage">You can also paste this traceback into
      a <a href="https://gist.github.com/">gist</a>:
      <input type="submit" value="create paste"></span>
    </p>
    <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py&quot;, line 2012, in delete
    state = attributes.instance_state(instance)
AttributeError: 'NoneType' object has no attribute '_sa_instance_state'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 2449, in wsgi_app
    response = self.handle_exception(e)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py&quot;, line 39, in reraise
    raise value
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py&quot;, line 39, in reraise
    raise value
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py&quot;, line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py&quot;, line 163, in dispatch_request
    return meth(*args, **kwargs)
  File &quot;C:\Users\admin\PycharmProjects\TrainingTask\app.py&quot;, line 112, in delete
    db.session.delete(data)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\scoping.py&quot;, line 162, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py&quot;, line 2015, in delete
    exc.UnmappedInstanceError(instance), replace_context=err,
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\util\compat.py&quot;, line 178, in raise_
    raise exception
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped</textarea>
  </form>
</div>
<div class="explanation">
  The debugger caught an exception in your WSGI application.  You can now
  look at the traceback which led to the error.  <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
</div>
      <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
      </div>
    </div>

    <div class="pin-prompt">
      <div class="inner">
        <h3>Console Locked</h3>
        <p>
          The console is locked and needs to be unlocked by entering the PIN.
          You can find the PIN printed out on the standard output of your
          shell that runs the server.
        <form>
          <p>PIN:
            <input type=text name=pin size=14>
            <input type=submit name=btn value="Confirm Pin">
        </form>
      </div>
    </div>
  </body>
</html>

<!--

Traceback (most recent call last):
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py", line 2012, in delete
    state = attributes.instance_state(instance)
AttributeError: 'NoneType' object has no attribute '_sa_instance_state'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py", line 163, in dispatch_request
    return meth(*args, **kwargs)
  File "C:\Users\admin\PycharmProjects\TrainingTask\app.py", line 112, in delete
    db.session.delete(data)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\scoping.py", line 162, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py", line 2015, in delete
    exc.UnmappedInstanceError(instance), replace_context=err,
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\util\compat.py", line 178, in raise_
    raise exception
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped

-->


21:09:14.270 xml parsing failed, response data type set to string: org.xml.sax.SAXParseException; lineNumber: 22; columnNumber: 5; The element type "link" must be terminated by the matching end-tag "</link>".
21:09:14.277 status code was: 500, expected: 200, response time: 162, url: http://127.0.0.1:5000/user/16, response: <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
  "http://www.w3.org/TR/html4/loose.dtd">
<html>
  <head>
    <title>sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css"
        type="text/css">
    <!-- We need to make sure this has a favicon so that the debugger does
         not by accident trigger a request to /favicon.ico which might
         change the application state. -->
    <link rel="shortcut icon"
        href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=jquery.js"></script>
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script type="text/javascript">
      var TRACEBACK = 2371604320664,
          CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "oiNCkKA32okpVbSfZqpA";
    </script>
  </head>
  <body style="background-color: #fff">
    <div class="debugger">
<h1>sqlalchemy.orm.exc.UnmappedInstanceError</h1>
<div class="detail">
  <p class="errormsg">sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped</p>
</div>
<h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
<div class="traceback">
  
  <ul><li><div class="frame" id="frame-2371606792736">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py"</cite>,
      line <em class="line">2012</em>,
      in <code class="function">delete</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws">        </span>if self._warn_on_events:</pre>
<pre class="line before"><span class="ws">            </span>self._flush_warning(&quot;Session.delete()&quot;)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line current"><span class="ws">            </span>state = attributes.instance_state(instance)</pre>
<pre class="line after"><span class="ws">        </span>except exc.NO_STATE as err:</pre>
<pre class="line after"><span class="ws">            </span>util.raise_(</pre>
<pre class="line after"><span class="ws">                </span>exc.UnmappedInstanceError(instance), replace_context=err,</pre>
<pre class="line after"><span class="ws">            </span>)</pre>
<pre class="line after"><span class="ws"></span> </pre></div>
</div>

<li><div class="exc-divider">The above exception was the direct cause of the following exception:</div>
<li><div class="frame" id="frame-2371606791000">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">2463</em>,
      in <code class="function">__call__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>def __call__(self, environ, start_response):</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;The WSGI server calls the Flask application object as the</pre>
<pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app` which can be</pre>
<pre class="line before"><span class="ws">        </span>wrapped to applying middleware.&quot;&quot;&quot;</pre>
<pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def __repr__(self):</pre>
<pre class="line after"><span class="ws">        </span>return &quot;&lt;%s %r&gt;&quot; % (self.__class__.__name__, self.name)</pre></div>
</div>

<li><div class="frame" id="frame-2371606791784">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">2449</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line before"><span class="ws">                </span>error = e</pre>
<pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre></div>
</div>

<li><div class="frame" id="frame-2371606792064">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1866</em>,
      in <code class="function">handle_exception</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span># if we want to repropagate the exception, we can attempt to</pre>
<pre class="line before"><span class="ws">            </span># raise it with the whole traceback in case we can do that</pre>
<pre class="line before"><span class="ws">            </span># (the function was actually called from the except part)</pre>
<pre class="line before"><span class="ws">            </span># otherwise, we just raise the error again</pre>
<pre class="line before"><span class="ws">            </span>if exc_value is e:</pre>
<pre class="line current"><span class="ws">                </span>reraise(exc_type, exc_value, tb)</pre>
<pre class="line after"><span class="ws">            </span>else:</pre>
<pre class="line after"><span class="ws">                </span>raise e</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>self.log_exception((exc_type, exc_value, tb))</pre>
<pre class="line after"><span class="ws">        </span>server_error = InternalServerError()</pre></div>
</div>

<li><div class="frame" id="frame-2371606793688">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py"</cite>,
      line <em class="line">39</em>,
      in <code class="function">reraise</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>import collections.abc as collections_abc</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>def reraise(tp, value, tb=None):</pre>
<pre class="line before"><span class="ws">        </span>if value.__traceback__ is not tb:</pre>
<pre class="line before"><span class="ws">            </span>raise value.with_traceback(tb)</pre>
<pre class="line current"><span class="ws">        </span>raise value</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>implements_to_string = _identity</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>else:</pre>
<pre class="line after"><span class="ws">    </span>iterkeys = lambda d: d.iterkeys()</pre></div>
</div>

<li><div class="frame" id="frame-2371606793464">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">2446</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
<pre class="line before"><span class="ws">        </span>error = None</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">                </span>error = e</pre>
<pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre></div>
</div>

<li><div class="frame" id="frame-2371606792288">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1951</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(self, rv, from_error_handler=False):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Given the return value from a view function this finalizes</pre>
<pre class="line after"><span class="ws">        </span>the request by converting it into a response and invoking the</pre></div>
</div>

<li><div class="frame" id="frame-2371606790832">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1820</em>,
      in <code class="function">handle_user_exception</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>return self.handle_http_exception(e)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>handler = self._find_error_handler(e)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>if handler is None:</pre>
<pre class="line current"><span class="ws">            </span>reraise(exc_type, exc_value, tb)</pre>
<pre class="line after"><span class="ws">        </span>return handler(e)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def handle_exception(self, e):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Handle an exception that did not have an error handler</pre>
<pre class="line after"><span class="ws">        </span>associated with it, or that was raised from an error handler.</pre></div>
</div>

<li><div class="frame" id="frame-2371606792960">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py"</cite>,
      line <em class="line">39</em>,
      in <code class="function">reraise</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>import collections.abc as collections_abc</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>def reraise(tp, value, tb=None):</pre>
<pre class="line before"><span class="ws">        </span>if value.__traceback__ is not tb:</pre>
<pre class="line before"><span class="ws">            </span>raise value.with_traceback(tb)</pre>
<pre class="line current"><span class="ws">        </span>raise value</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>implements_to_string = _identity</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>else:</pre>
<pre class="line after"><span class="ws">    </span>iterkeys = lambda d: d.iterkeys()</pre></div>
</div>

<li><div class="frame" id="frame-2371606793352">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1949</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>self.try_trigger_before_first_request_functions()</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line current"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(self, rv, from_error_handler=False):</pre></div>
</div>

<li><div class="frame" id="frame-2371606790384">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py"</cite>,
      line <em class="line">1935</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>getattr(rule, &quot;provide_automatic_options&quot;, False)</pre>
<pre class="line before"><span class="ws">            </span>and req.method == &quot;OPTIONS&quot;</pre>
<pre class="line before"><span class="ws">        </span>):</pre>
<pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
<pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
<pre class="line current"><span class="ws">        </span>return self.view_functions[rule.endpoint](**req.view_args)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def full_dispatch_request(self):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Dispatches the request and on top of that performs request</pre>
<pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
<pre class="line after"><span class="ws">        </span>error handling.</pre></div>
</div>

<li><div class="frame" id="frame-2371606791952">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py"</cite>,
      line <em class="line">89</em>,
      in <code class="function">view</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>constructor of the class.</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>def view(*args, **kwargs):</pre>
<pre class="line before"><span class="ws">            </span>self = view.view_class(*class_args, **class_kwargs)</pre>
<pre class="line current"><span class="ws">            </span>return self.dispatch_request(*args, **kwargs)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>if cls.decorators:</pre>
<pre class="line after"><span class="ws">            </span>view.__name__ = name</pre>
<pre class="line after"><span class="ws">            </span>view.__module__ = cls.__module__</pre>
<pre class="line after"><span class="ws">            </span>for decorator in cls.decorators:</pre></div>
</div>

<li><div class="frame" id="frame-2371606792624">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py"</cite>,
      line <em class="line">163</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span># retry with GET.</pre>
<pre class="line before"><span class="ws">        </span>if meth is None and request.method == &quot;HEAD&quot;:</pre>
<pre class="line before"><span class="ws">            </span>meth = getattr(self, &quot;get&quot;, None)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>assert meth is not None, &quot;Unimplemented method %r&quot; % request.method</pre>
<pre class="line current"><span class="ws">        </span>return meth(*args, **kwargs)</pre></div>
</div>

<li><div class="frame" id="frame-2371606792008">
  <h4>File <cite class="filename">"C:\Users\admin\PycharmProjects\TrainingTask\app.py"</cite>,
      line <em class="line">112</em>,
      in <code class="function">delete</code></h4>
  <div class="source "><pre class="line before"><span class="ws">                  </span>type: array</pre>
<pre class="line before"><span class="ws">                  </span>items:</pre>
<pre class="line before"><span class="ws">                    </span>$ref: '#/definitions/User'</pre>
<pre class="line before"><span class="ws">                </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws">        </span>data = UserModel.query.get(user_id)</pre>
<pre class="line current"><span class="ws">        </span>db.session.delete(data)</pre>
<pre class="line after"><span class="ws">        </span>db.session.commit()</pre>
<pre class="line after"><span class="ws">        </span>return jsonify(UserModelSchema().dump(data))</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def put(self, user_id):</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;</pre></div>
</div>

<li><div class="frame" id="frame-2371606791392">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\scoping.py"</cite>,
      line <em class="line">162</em>,
      in <code class="function">do</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span>&quot;&quot;&quot;Old name for backwards compatibility.&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws"></span>def instrument(name):</pre>
<pre class="line before"><span class="ws">    </span>def do(self, *args, **kwargs):</pre>
<pre class="line current"><span class="ws">        </span>return getattr(self.registry(), name)(*args, **kwargs)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>return do</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>for meth in Session.public_methods:</pre></div>
</div>

<li><div class="frame" id="frame-2371606791448">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py"</cite>,
      line <em class="line">2015</em>,
      in <code class="function">delete</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>state = attributes.instance_state(instance)</pre>
<pre class="line before"><span class="ws">        </span>except exc.NO_STATE as err:</pre>
<pre class="line before"><span class="ws">            </span>util.raise_(</pre>
<pre class="line current"><span class="ws">                </span>exc.UnmappedInstanceError(instance), replace_context=err,</pre>
<pre class="line after"><span class="ws">            </span>)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>self._delete_impl(state, instance, head=True)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def _delete_impl(self, state, obj, head):</pre></div>
</div>

<li><div class="frame" id="frame-2371606790496">
  <h4>File <cite class="filename">"C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\util\compat.py"</cite>,
      line <em class="line">178</em>,
      in <code class="function">raise_</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span># intermediary exception does not change, but we can't figure</pre>
<pre class="line before"><span class="ws">            </span># that out.</pre>
<pre class="line before"><span class="ws">            </span>exception.__cause__ = replace_context</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line current"><span class="ws">            </span>raise exception</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre>
<pre class="line after"><span class="ws">            </span># credit to</pre>
<pre class="line after"><span class="ws">            </span># https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/</pre>
<pre class="line after"><span class="ws">            </span># as the __traceback__ object creates a cycle</pre>
<pre class="line after"><span class="ws">            </span>del exception, replace_context, from_, with_traceback</pre></div>
</div>
</ul>
  <blockquote>sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped</blockquote>
</div>

<div class="plain">
  <form action="/?__debugger__=yes&amp;cmd=paste" method="post">
    <p>
      <input type="hidden" name="language" value="pytb">
      This is the Copy/Paste friendly version of the traceback.  <span
      class="pastemessage">You can also paste this traceback into
      a <a href="https://gist.github.com/">gist</a>:
      <input type="submit" value="create paste"></span>
    </p>
    <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py&quot;, line 2012, in delete
    state = attributes.instance_state(instance)
AttributeError: 'NoneType' object has no attribute '_sa_instance_state'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 2449, in wsgi_app
    response = self.handle_exception(e)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py&quot;, line 39, in reraise
    raise value
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py&quot;, line 39, in reraise
    raise value
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py&quot;, line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py&quot;, line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py&quot;, line 163, in dispatch_request
    return meth(*args, **kwargs)
  File &quot;C:\Users\admin\PycharmProjects\TrainingTask\app.py&quot;, line 112, in delete
    db.session.delete(data)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\scoping.py&quot;, line 162, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py&quot;, line 2015, in delete
    exc.UnmappedInstanceError(instance), replace_context=err,
  File &quot;C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\util\compat.py&quot;, line 178, in raise_
    raise exception
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped</textarea>
  </form>
</div>
<div class="explanation">
  The debugger caught an exception in your WSGI application.  You can now
  look at the traceback which led to the error.  <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
</div>
      <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
      </div>
    </div>

    <div class="pin-prompt">
      <div class="inner">
        <h3>Console Locked</h3>
        <p>
          The console is locked and needs to be unlocked by entering the PIN.
          You can find the PIN printed out on the standard output of your
          shell that runs the server.
        <form>
          <p>PIN:
            <input type=text name=pin size=14>
            <input type=submit name=btn value="Confirm Pin">
        </form>
      </div>
    </div>
  </body>
</html>

<!--

Traceback (most recent call last):
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py", line 2012, in delete
    state = attributes.instance_state(instance)
AttributeError: 'NoneType' object has no attribute '_sa_instance_state'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\flask\views.py", line 163, in dispatch_request
    return meth(*args, **kwargs)
  File "C:\Users\admin\PycharmProjects\TrainingTask\app.py", line 112, in delete
    db.session.delete(data)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\scoping.py", line 162, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\orm\session.py", line 2015, in delete
    exc.UnmappedInstanceError(instance), replace_context=err,
  File "C:\Users\admin\AppData\Local\Programs\Python\Python36\lib\site-packages\sqlalchemy\util\compat.py", line 178, in raise_
    raise exception
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped

-->

21:09:14.315 karate.env system property was:  
21:09:14.316 request:
1 > GET http://127.0.0.1:5000/users
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Host: 127.0.0.1:5000
1 > User-Agent: Apache-HttpClient/4.5.11 (Java/1.8.0_241)

21:09:14.351 response time in milliseconds: 30.89
1 < 200
1 < Content-Length: 1353
1 < Content-Type: application/json
1 < Date: Mon, 30 Mar 2020 15:39:14 GMT
1 < Server: Werkzeug/0.16.1 Python/3.6.6
[
  {
    "email": "balaji@gmail.com", 
    "id": 1, 
    "name": "balaji", 
    "password": "sai"
  }, 
  {
    "email": "ram@gmail.com", 
    "id": 2, 
    "name": "ram", 
    "password": "sai"
  }, 
  {
    "email": "Lusy12@gmail.com", 
    "id": 3, 
    "name": "Lusy", 
    "password": "1234we"
  }, 
  {
    "email": "john@gmail.com", 
    "id": 6, 
    "name": "John Smith", 
    "password": "balaji"
  }, 
  {
    "email": "jack@gmail.com", 
    "id": 8, 
    "name": "jack", 
    "password": "jack"
  }, 
  {
    "email": "john@gmail.com", 
    "id": 9, 
    "name": "john", 
    "password": "john"
  }, 
  {
    "email": "mary@gmail.com", 
    "id": 13, 
    "name": "mary", 
    "password": "mary"
  }, 
  {
    "email": "jackson@gmail.com", 
    "id": 14, 
    "name": "jackson", 
    "password": "jackson"
  }, 
  {
    "email": "henry@gmail.com", 
    "id": 15, 
    "name": "henry", 
    "password": "henry"
  }, 
  {
    "email": "magrit@gmail.com", 
    "id": 17, 
    "name": "magrit", 
    "password": "henry"
  }, 
  {
    "email": "harley@gmail.com", 
    "id": 18, 
    "name": "harley", 
    "password": "henry"
  }, 
  {
    "email": "harley@gmail.com", 
    "id": 19, 
    "name": "harley", 
    "password": "henry"
  }, 
  {
    "email": "harley@gmail.com", 
    "id": 20, 
    "name": "harley", 
    "password": "henry"
  }
]


21:09:14.364 karate.env system property was:  
21:09:14.374 request:
1 > GET http://127.0.0.1:5000/user/2
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Host: 127.0.0.1:5000
1 > User-Agent: Apache-HttpClient/4.5.11 (Java/1.8.0_241)

21:09:14.396 response time in milliseconds: 21.87
1 < 200
1 < Content-Length: 83
1 < Content-Type: application/json
1 < Date: Mon, 30 Mar 2020 15:39:14 GMT
1 < Server: Werkzeug/0.16.1 Python/3.6.6
{
  "email": "ram@gmail.com", 
  "id": 2, 
  "name": "ram", 
  "password": "sai"
}


21:09:14.436 karate.env system property was:  
21:09:14.446 request:
1 > PUT http://127.0.0.1:5000/user/13
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Content-Length: 58
1 > Content-Type: application/json; charset=UTF-8
1 > Host: 127.0.0.1:5000
1 > User-Agent: Apache-HttpClient/4.5.11 (Java/1.8.0_241)
{"name":"mary","email":"mary@gmail.com","password":"mary"}

21:09:14.463 response time in milliseconds: 16.37
1 < 200
1 < Content-Length: 25
1 < Content-Type: text/html; charset=utf-8
1 < Date: Mon, 30 Mar 2020 15:39:14 GMT
1 < Server: Werkzeug/0.16.1 Python/3.6.6
User Updated Successfully

